fastpair/rust/src/bluetooth/windows_ble/adapter.rs

```diff
@@ -35,6 +73,87 @@ impl BleAdapter {
 
         Ok(BleAdapter { inner })
     }
+
+    pub fn scanner(&self) -> Result<impl Stream<Item = BleDevice>, anyhow::Error> {
+        let watcher = BluetoothLEAdvertisementWatcher::new()?;
+        watcher.SetScanningMode(BluetoothLEScanningMode::Active)?;
+
+        if self.inner.IsExtendedAdvertisingSupported()? {
+            watcher.SetAllowExtendedAdvertisements(true)?;
+        }
+
+        let (sender, receiver) = futures::channel::mpsc::channel(16);
+        let sender = Arc::new(std::sync::Mutex::new(sender));
```

Comment:
why is this needed?

Reply:
nvm

Reply:
Added comment to prevent further confusion.